tvm/python/tvm/topi/adreno/conv2d_nhwc.py

```diff
@@ -186,38 +242,34 @@ def schedule_conv2d_NHWC(cfg, s, output, args={}):
     cfg.multi_filter(filter=lambda entity: entity["tile_fc"].size[2] * entity["tile_y"].size[2] * entity["tile_x"].size[2] in range(32,1024))
     ##### space definition end #####
 
-    #if autotvm.GLOBAL_SCOPE.in_tuning:
-    #    pad_data, flattened_kernel = s[conv].op.input_tensors
-    #    kernel = s[flattened_kernel].op.input_tensors[0]
-    #    s[flattened_kernel].compute_inline()
-    #else:
-    #    pad_data, kernel = s[conv].op.input_tensors
-    #    flattened_kernel = kernel
-
     pad_data, kernel = s[conv].op.input_tensors
-    s[kernel].compute_inline()
+    pack_data = pad_data.op.input_tensors[0]
+    if s[pack_data].op.tag != "input_pack_expanded":
+        s[pack_data].compute_inline()
+    else:
+        axes = s[pack_data].op.axis
+        fused = s[pack_data].fuse(*axes[:-1])
+        shape = get_const_tuple(pack_data.shape)
+        ftc = numpy.prod(shape[:-1])
+        div = getDiv(ftc, 64)
```

Comment:
could you please provide comment on the reason for choosing 64? Is it workgroup or warp size?

Reply:
By fact this parameter must be a subject for tuning as well. Unfortunately adding three more parameters to the current schedules will increase search space even more. Even right now there are millions combinations to measure. If we add 3 more parameters each of them can be represented by even several numbers, the seach space might be increased to several hundreds millions variants.
So far I decided not to add it to the search space but hard code them based on my manual measurements. On several standalone convolutions I saw that 64 is the best choice in this certain place (for repacking of inputs). In other place I saw that 128 is the best choice.